rfcs/0000-commonmark-docs.md

```diff
@@ -0,0 +1,302 @@
+demo below, it was easy to fix 5 bugs already.
+
+We propose to keep the man pages as Docbook for now. They are
+self-contained documents, whose form is constrained by convention and
+the limits of the man page format. When the time comes to convert the
+man pages as well, we can turn here again to prior art. The Kurnetes
+project uses [md2man][md2man] to generate man pages from CommonMark.
+This is a small Go command with a 4.1MB closure size (including 2MB
+for `tzdata`).
+
+[gatsby]: https://gatsbyjs.org
+[sphinx]: https://www.sphinx-doc.org
+[pandoc]: https://pandoc.org/
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+This RFC is *not* about vetting a single implementation choice. The
+five large projects cited above use one of Hugo, Gatsby, Jekyll, or
+mdBook. We can add Sphinx to the set of available implementation
+choices, which likewise has excellent support for Markdown. In this
+section we show how two of these choices could work in the context of
+the Nix* projects. *Selecting a toolchain is left to the discretion of
+the implementors. Discussing implementation choices that are
+transparent to the user is out of scope for an RFC process*.
+
+The following demo shows how to satisfy all requirements above using
+CommonMark as the input format and a Gatsby-based [starter kit from
+Hasura][hasura-docs-starter] to generate a ready-to-deploy static
+website:
+
+https://nixos-docs-mockup.netlify.app
+
+It features the following sections of the Nix manual:
+* introduction
+* quickstart and installation
+* the Nix experssions chapter
+* advanced topics
+
+The content is the unedited output of running the Pandoc conversion,
+which still has bugs (in particular the handling of cross-references).
+
+This other demo shows how to satisfy many of the same requirements
+using Sphinx, again with CommonMark as the input:
+
+https://nixpkgs-manual-sphinx-markedown-example.netlify.app
+
+The look and feel is customizable, and indeed could be made the same
+in both cases. Both are examples meant to demonstrate that choosing
+CommonMark as the input format doesn't force unreasonable compromises.
+
+[hasura-docs-starter]: https://github.com/hasura/gatsby-gitbook-starter
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The current DocBook format is semantically richer. There are specific
+tags for definitions, environment variables, user accounts, various
+types of callouts, etc. CommonMark's data model isn't nearly as rich,
+so in converting to CommonMark, some information is lost. However,
+experience in other very large ecosystems with many users tells us
+that authors are happy to make do with an inexpressive but familiar
+format, which in any case *can* be extended with the wise use of
+`<span>`-like HTML tags and custom tags that expand to HTML. That
+appears to be seldom necessary in practice (see e.g. the five
+documentation examples in [Motivation][motivation]).
```

Comment:
Drawback: no support for PDF, epub or manual pages.

Reply:
We don't need PDF or epub. Manpages are apparently supported by Sphinx.

Reply:
So the tooling matters here again, otherwise we will have to ditch `services.nixosManual.showManual` option.

Reply:
Why? That option only requires an HTML rendering of the manual.

Reply:
Ah right, I forgot that uses HTML instead of the manual page.